pandas/core/indexes/datetimes.py

```diff
@@ -394,11 +394,17 @@ def _is_dates_only(self) -> bool:
         -------
         bool
         """
+        from pandas import Timedelta
+
         from pandas.io.formats.format import is_dates_only
 
         # error: Argument 1 to "is_dates_only" has incompatible type
         # "Union[ExtensionArray, ndarray]"; expected "Union[ndarray,
         # DatetimeArray, Index, DatetimeIndex]"
+        delta = self.freq.delta if self.freq and hasattr(self.freq, "delta") else None
+
+        if delta and delta < Timedelta(days=1):
```

Comment:
`if delta and delta %  pd.Timedelta(days=1) == pd.Timedelta(days=0):`

Would catch all timedeltas that wouldn't match up to days. Seems better?

This should also go above the comment, as the coment concerns line 408.

Reply:
I'm not sure if that would work. Reason being is that if you create a datetime index like this:

`DatetimeIndex(["2012-01-01 00:00:00"],freq=str(60)+'T')`

Then the conditional you proposed would not be true. 

Also, I think this goes back to what @pmlpm1986 was saying with how the crux of the problem is that a datetime is being recognized as a date. However, you mentioned this fix [here ](https://github.com/pandas-dev/pandas/issues/53470#issuecomment-1579502567)so I decided to implement it.

Reply:
Yeah, `==` should have been `!=` in my example, then it should work.

Can you check again? This will cover cases like `pd.DatetimeIndex(["2012-01-01 00:00:00"],freq='25H')`, which isn't covered in the first example.

Reply:
I think there should be a test that explicitly covers those cases (i.e., 25 or 36 hour frequency with a single entry). That way, we ensure the behaviour is consistent and that the solution works.

Reply:
Yes, can you add them to the tests you've written (and rename the tests to signify it tests non-day freqs).

Reply:
I have a added to Simar-B's PR/fork to include more tests. I added comments to make sure they can are easily understood.

Reply:
I didn't see that it wasn't @Simar-B who was responding, sorry, usually it is the PR's owner who adds the commits. Also I don't see any update to the PR?